lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -32,6 +33,8 @@ static LogicalResult checkStructure(linalg::LinalgOp linalgOp) {
   if (iteratorTypes.size() < 4)
     return failure();
   size_t size = iteratorTypes.size() - 1;
+  if (size == 6)
```

Comment:
Would it make sense to split the logic out of this function? `checkStructure` was specific for linalg.brgemm

Reply:
Also, we shouldn't rely on just the size for this check. This needs to make sure the extra dimension is really a VNNI one (dim/type = 2/bf16, 4/bf8 and coming from a pack etc).

Reply:
It's not always possible to differentiate a packed `<NxMxnxTy>` from a standard `<MxNxKxTy>` where both `n` and `K` are 2.

the main tell-tales are:
* inner dim is 2 and type is BF16 || inner dim is 4 and type is bf8
* the _other_ shape in the operation doesn't have the same dimension (is this always true?)
* the producer is a `pack` operation with VNNI shape at some point in the use-def chain (this is hard to check)
* none of the ops in the use-def chain undo the VNNI pack (like matmul) (also hard to check)

In the end, it's not a trivial check, so perhaps we could keep a set somewhere of all packed shapes, adding them when we issue a `pack` and removing when we issue an `unpack` or a `matmul` that we know undoes the pack.

A quick check on that set would tell us that the shape is packed, and looking at the inner dimension + type should tell us which way.